### ISSUE TITLE
Slang Failed to reflect SPIR-V with CRT-Royale

### ISSUE THREAD

**Issue by billywade** [libretro/RetroArch#8049]
# First and foremost consider this:
- Only RetroArch bugs should be filed here. Not core bugs or game bugs
- This is not a forum or a help section, this is strictly developer oriented

## Description

When applying any of the CRT-Royale shaders, nothing changes on screen. After looking at the log, it reads:
```
[INFO] [slang]: Compiling shader "/home/trey/.config/retroarch/shaders/shaders_slang/crt/shaders/crt-royale/src/crt-royale-bloom-approx.slang".
[INFO] [slang]: Using render target format R8G8B8A8_SRGB for pass output #0.
[INFO] [slang]: Building pass #0 (N/A)
[INFO] [Vulkan filter chain]: Creating framebuffer 1024 x 1024 (max 1 level(s)).
[ERROR] [slang]: Non-semantic textures not supported yet.
[ERROR] [slang]: Failed to reflect SPIR-V. Resource usage is inconsistent with expectations.
[ERROR] [Vulkan]: Failed to create preset: "/home/user/.config/retroarch/shaders/retroarch-psx.slangp".
[ERROR] [Vulkan]: Failed to create filter chain: "/home/user/.config/retroarch/shaders/retroarch-psx.slangp". Falling back to stock.
```

### Expected behavior

Vulkan to compile and display the shader.

### Actual behavior

Something is breaking SPIR-V.

### Steps to reproduce the bug

1. Change to Vulkan and use a Vulkan hardware core (In this case it was Beetle PSX HW).
2. Turn on any of the CRT-Royale shaders under Slang.

### Version/Commit
You can find this information under Information/System Information

- RetroArch: 1.7.5

### Environment information

- OS: Manjaro Linux 64-bit, up-to-date as of yesterday.

**Comment by orbea** [comment]
I can't reproduce this here with an amd rx vega 56 and radv.

What graphics card do you have?
Can you try a nightly version?
Can you try updating the slang shaders from the buildbot?

**Comment by orbea** [comment]
I suppose this might be a nvidia vulkan bug, do other slang shaders work at all?

**Comment by billywade** [comment]
Yeah, most texture-focused shaders like xBR and Eagle are fine, and Easymode and Geom work.

**Comment by billywade** [comment]
That's what I was worried about. The thing is, though, some of the people on the dev team have it working [here](https://www.libretro.com/index.php/new-vulkan-graphics-enhancements-features-in-beetle-psx-hw/).

**Comment by orbea** [comment]
My guess that its the difference in graphics cards or even nvidia driver versions. Looking at the code I guess it would work if Non-semantic textures are supported in RetroArch and those checks are removed, but I don't even know what those are...

You could try removing the errors and checks and see where it blows up or not...

**Comment by billywade** [comment]
I might do that, but it feels a little low level, especially if it's just a matter of using the right config for Royale. I'll see if I can get a hold of Daniel and ask him how he got it working.

**Comment by hizzlekizzle** [comment]
The "non-semantic texture" error typically happens when a pass refers to a texture/alias that doesn't exist in the preset.

Do the royale presets work for you with other cores?

**Comment by twinaphex** [comment]
@Themaister any thoughts here?

**Comment by billywade** [comment]
@hizzlekizzle no, but I'd guess that the reason depends on the video driver used. It has the same effect on paraLLEl, and it causes a black screen on SNES9x and Mupen64Plus, although with the latter there's a slowdown at higher passes, so I figure that's a configuration problem rather than a problem with the shader itself.

**Comment by Themaister** [comment]
Non-semantic texture refers to that a texture has been declared with a name which RA does not understand. Maybe it's referring to a LUT which is not part of the .slangp?

**Comment by billywade** [comment]
Update: Using a fresh config file has no effect.

**Comment by orbea** [comment]
What option in the fresh config file was giving you a black screen?

**Comment by billywade** [comment]
@oreba it was my problem, not RetroArch; I forgot to set the resources folder, so Beetle didn't have a BIOS to use.

**Comment by hizzlekizzle** [comment]
And I assume you've tried updating the slang shaders via the online updater?

**Comment by orbea** [comment]
@hizzlekizzle 

> * Can you try updating the slang shaders from the buildbot?
Just tried it, also no luck. It's still reporting the same errors. However, now I'm getting this weird thing in the corner. My guess is that it's because I built DEBUG=1 in case you needed a backtrace, but I've never seen it before, even other times I've built debug.

https://github.com/libretro/RetroArch/issues/8049#issuecomment-455880109

**Comment by billywade** [comment]
The error in that also was just a weird thing Nvidia was doing, not anything in the shaders.

**Comment by hizzlekizzle** [comment]
It seems this error can also come up if any of the parameter pragmas are malformed.
Are you sure you the shader files and presets haven't been modified in any way?

**Comment by orbea** [comment]
Try entirely removing the shaders from your config directory and re-downloading them to be sure.

**Comment by billywade** [comment]
Will do. Is there a specific way I need to use the buildbot? When I try using the webclient it's shockingly slow.

**Comment by orbea** [comment]
Just `mv` or `rm` your old shaders in your config directory and then download them with the online updater.

**Comment by billywade** [comment]
Alright, just deleted everything and downloaded from the updater, and nothing still.

**Comment by billywade** [comment]
where would I find that at? I couldn't find it in either settings files.

**Comment by hizzlekizzle** [comment]
it's in the slangp preset, for example:
https://github.com/libretro/slang-shaders/blob/master/crt/crt-royale.slangp#L19
